src/mapgen_functions.cpp

```diff
@@ -4167,3 +4113,72 @@ void madd_field( map *m, int x, int y, field_id t, int density )
     tripoint actual_location( x, y, m->get_abs_sub().z );
     m->add_field( actual_location, t, density, 0 );
 }
+
+void place_stairs( map *m, oter_id terrain_type, mapgendata dat,
+                   const int actual_house_height, const int lw, const int rw )
+{
+    if( dat.has_basement() ) {
```

Comment:
Since you moved this into it's own function you can drop an entire level of indentation by doing an early return with `if( !dat.has_basement() ) return;`